src/test/lib/operators/import_test.cpp

```diff
@@ -18,55 +27,215 @@ class OperatorsImportTest : public BaseTest {
+
+std::string file_types_and_encodings_name_generator(
+    const ::testing::TestParamInfo<std::tuple<FileType, std::optional<EncodingType>>>& info) {
+  const auto& [file_type, encoding_type] = info.param;
+
+  // Convert FileType and EncodingType to strings for a human readable name.
+  const std::string file_type_str = std::string{magic_enum::enum_name(file_type)};
+  const std::string encoding_type_str = encoding_type ? std::string{magic_enum::enum_name(*encoding_type)} : "None";
 
-INSTANTIATE_TEST_SUITE_P(FileTypes, OperatorsImportMultiFileTypeTest,
-                         ::testing::Values(FileType::Csv, FileType::Tbl, FileType::Binary), enum_formatter<FileType>);
+  return file_type_str + "_" + encoding_type_str;
+}
+
+// Both FixedStringDictionary and FrameOfReference do not work with float. They are tested with int and string below.
+INSTANTIATE_TEST_SUITE_P(FileTypesAndEncodings, OperatorsImportMultiFileTypeAndEncodingTest,
+                         ::testing::Combine(::testing::Values(FileType::Csv, FileType::Tbl, FileType::Binary),
+                                            ::testing::Values(std::nullopt, EncodingType::Unencoded,
+                                                              EncodingType::Dictionary, EncodingType::RunLength,
+                                                              EncodingType::LZ4)),
+                         file_types_and_encodings_name_generator);
 
-TEST_P(OperatorsImportMultiFileTypeTest, ImportWithFileType) {
+TEST_P(OperatorsImportMultiFileTypeAndEncodingTest, ImportEncodingAndFileType) {
   auto expected_table =
       std::make_shared<Table>(TableColumnDefinitions{{"a", DataType::Float, false}}, TableType::Data, ChunkOffset{5});
   expected_table->append({1.1f});
   expected_table->append({2.2f});
   expected_table->append({3.3f});
   expected_table->append({4.4f});
 
-  const auto reference_filename = reference_filepath + reference_filenames.at(GetParam());
-  auto importer = std::make_shared<Import>(reference_filename, "a", Chunk::DEFAULT_SIZE, GetParam());
+  // Make all chunks in the expected table immutable to allow encoding.
+  for (auto chunk_id = ChunkID{0}; chunk_id < expected_table->chunk_count(); ++chunk_id) {
+    expected_table->get_chunk(chunk_id)->set_immutable();
+  }
+
+  const auto& [file_type, encoding] = GetParam();
+
+  // Apply the encoding to the table.
+  if (encoding) {
+    const auto encoding_spec = SegmentEncodingSpec{*encoding};
+    ChunkEncoder::encode_all_chunks(expected_table, encoding_spec);
+  } else if (file_type != FileType::Binary) {
+    // If no encoding is specified we want to check if the defaults are as expected.
+    // The binary file is unencoded so we do not have to change the encoding of `expected_table`.
+    // For `.csv` and `.tbl` the default encoding should be `EncodingType::Dictionary`.
+    const auto encoding_spec = SegmentEncodingSpec{EncodingType::Dictionary};
```

Comment:
`SegmentEncodingSpec{}` even uses `EncosingType::Dictionary` as the default:
https://github.com/hyrise/hyrise/blob/1d7f8607ab004922e63d6ab55b4f8986116c6e5c/src/lib/storage/encoding_type.hpp#L61